### PR TITLE
Equivalence checker helper for Act

### PR DESCRIPTION
**Pull request by zoep**
## Description

Creates an equivalence check helper function that can be used by Act to check equivalence between two Expr programs. 

## Checklist

- [X] tested locally
- [ ] added automated tests
- [ ] updated the docs
- [ ] updated the changelog

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -32,7 +32,7 @@
           z3
           cvc5
           git
-          foundry.defaultPackage.${system}
+  #       foundry.defaultPackage.${system}
```
Comment (d-xo) [inline flake.nix:35]: please leave this uncommented on main

Comment (zoep) [reply flake.nix]: Done!

File: flake.nix [hunk 2]
```diff
@@ -130,6 +130,7 @@
         # --- packages ----
 
         packages.withTests = hevmUnwrapped;
+	packages.noTests = pkgs.haskell.lib.dontCheck hevmUnwrapped;
```
[the remark this answers is not in the document]
Comment (zoep) [reply flake.nix]: Done!

**Review by d-xo — APPROVED** [review]
looks good!

**Pull request merged**